added covariance matrices to HP_2011_ds62 and JH_2015

bobmyhill opened the pull request:
The Holland and Powell databases include a full variance-covariance matrix for the free energies of formation, derived from the least-squares fitting of the experimental data. This PR adds the matrix to the module burnman.minerals.HP_2011_ds62, as a dictionary containing:
- 'endmember_names', an ordered list of all the endmember names as provided in each endmember param dictionary, and 
- 'covariance_matrix': the covariance matrix

JH_2015 also now includes a covariance matrix derived from HP_2011_ds62, but including all the CombinedMinerals.

bobmyhill commented:
/rebuild

tjhei commented on burnman/data/input_raw_endmember_datasets/HPdata_to_burnman.py:
```diff
@@ -74,74 +68,105 @@ def getmbr(ds, mbr):
-        print ''
-print ''
+with open('HP_2011_ds62.py', 'wb') as outfile:
+    outfile.write('# This file is part of BurnMan - a thermoelastic and '
```
wouldn't it be cleaner if you use a python multiline string ``"""    """``?

bobmyhill replied:
That makes for really ugly looking code (as I would not be able to indent consecutive lines - the multiline string includes any blank space before or after the code)

bobmyhill replied:
(c.f. https://stackoverflow.com/questions/10660435/pythonic-way-to-create-a-long-multi-line-string)

tjhei commented on burnman/data/input_raw_endmember_datasets/HPdata_to_burnman.py:
```diff
@@ -74,74 +68,105 @@ def getmbr(ds, mbr):
+                          '            \'formula\': formula,\n'
+                          '            \'equation_of_state\': \'hp_tmt\',\n'
+                          '            \'H_0\': {0},'.format(M.H * 1e3)+'\n'
+                          '            \'S_0\': {0},'.format(M.S * 1e3)+'\n'
```
this also looks more complicated than it should. Can you not construct the dictionary natively and pretty-print it?

bobmyhill replied:
I don't know whether there's an easy way to get the indenting correct in the output code. This was the neatest solution I could find without pickling (which can't deal with classes anyway).

tjhei commented on burnman/minerals/JH_2015.py:
```diff
@@ -198,30 +223,80 @@ class fm(CombinedMineral):
-        
+
+"""
+VARIANCE-COVARIANCE MATRIX
```
this documentation won't be attached to ``cov``

tjhei commented on burnman/minerals/JH_2015.py:
```diff
@@ -198,30 +223,80 @@ class fm(CombinedMineral):
+
+cov['covariance_matrix'] = A.dot(HP_2011_ds62.cov['covariance_matrix']).dot(A.T)
+
+del A
```
wouldn't it be neater to create a function and call it to fill ``cov`` instead of executing code and deleting variables?

bobmyhill commented:
\rebuild

bobmyhill commented:
Hi @tjhei ... I think I've addressed all your comments now. For some reason the tester isn't running on this one any more, but I think it's ready to go.

tjhei commented:
can you rebase to master? This should make it run.

bobmyhill commented:
Ah, I see! Thanks :)

bobmyhill commented:
Thanks @tjhei :)
